app/models/user.rb

```diff
@@ -81,6 +84,25 @@ class User < ActiveRecord::Base
     end
   }
 
+  def require_2sv?
+    return false if otp_secret_key.present?
+
+    if deferred_2sv_at?
+      deferred_2sv_at < 24.hours.ago
+    else
+      super
+    end
+  end
+
+  def defer_two_step_verification
+    transaction do
+      self.require_2sv = false
```

Comment:
Ditto here, setting the `deferred_2sv_at` attribute should override the `require_2sv` flag. I feel `require_2sv` should be an intent from admins/higher ups that says "This user should, at some point, set up 2SV"